crates/e2e-test-utils/src/node.rs

```diff
@@ -1,37 +1,42 @@
-use crate::{engine_api::EngineApiHelper, network::NetworkHelper, payload::PayloadHelper};
+use crate::{
+    engine_api::EngineApiHelper, network::NetworkHelper, payload::PayloadHelper,
+    traits::PayloadEnvelopeV3,
+};
 use alloy_rpc_types::BlockNumberOrTag;
 use eyre::Ok;
 use reth::{
-    api::FullNodeComponents,
+    api::{BuiltPayload, EngineTypes, FullNodeComponents, PayloadBuilderAttributes},
     builder::FullNode,
     providers::{BlockReaderIdExt, CanonStateSubscriptions},
     rpc::{
         eth::{error::EthResult, EthTransactions},
         types::engine::PayloadAttributes,
     },
 };
-
-use reth_node_ethereum::EthEngineTypes;
 use reth_payload_builder::EthPayloadBuilderAttributes;
-use reth_primitives::{Address, Bytes, B256};
-
-use std::time::{SystemTime, UNIX_EPOCH};
+use reth_primitives::{Address, BlockNumber, Bytes, B256};
+use std::{
+    marker::PhantomData,
+    time::{SystemTime, UNIX_EPOCH},
+};
 use tokio_stream::StreamExt;
 
 /// An helper struct to handle node actions
-pub struct NodeHelper<Node>
+pub struct NodeHelper<Node, EngineType>
 where
-    Node: FullNodeComponents<Engine = EthEngineTypes>,
+    Node: FullNodeComponents<Engine = EngineType>,
+    EngineType: EngineTypes + 'static,
```

Comment:
hmm is this additional generic required here?
FullNodeComponents already has this?